src/blockdev.rs

```diff
@@ -1452,12 +1460,13 @@ mod tests {
 
         // test merging with unformatted initial disk
         let mut disk = make_unformatted_disk();
-        let saved = SavedPartitions::new_from_file(&mut disk, 512, &vec![label("z")]).unwrap();
-        let mut disk = make_disk(512, &merge_base_parts);
-        saved.merge(&mut image, &mut disk).unwrap();
-        let result = GPT::find_from(&mut disk).unwrap();
-        assert_partitions_eq(&merge_base_parts, &result, "unformatted disk");
-
+        let err = SavedPartitions::new_from_file(&mut disk, 512, &vec![label("z")]).unwrap_err();
```

Comment:
This is an expectation change, in the past, we would say there is no GPT, and then forget the list of saved partitions. 

Now that we are reacting to no GPT or MBR, we get in a state where we dont want to proceed. 

The user thinks there are partitions to save, but the filesystem is not known, as a result we cannot continue, and keep the promise of saving the partitions. so we bail, does that makes sense?